pageserver/src/tenant/storage_layer/layer.rs

```diff
@@ -129,19 +129,16 @@ pub(crate) fn local_layer_path(
     tenant_shard_id: &TenantShardId,
     timeline_id: &TimelineId,
     layer_file_name: &LayerName,
-    _generation: &Generation,
+    generation: &Generation,
 ) -> Utf8PathBuf {
     let timeline_path = conf.timeline_path(tenant_shard_id, timeline_id);
 
-    timeline_path.join(layer_file_name.to_string())
-
-    // TODO: switch to enabling new-style layer paths after next release
-    // if generation.is_none() {
-    //     // Without a generation, we may only use legacy path style
-    //     timeline_path.join(layer_file_name.to_string())
-    // } else {
-    //     timeline_path.join(format!("{}-v1{}", layer_file_name, generation.get_suffix()))
-    // }
+    if generation.is_none() {
+        // Without a generation, we may only use legacy path style
+        timeline_path.join(layer_file_name.to_string())
+    } else {
+        timeline_path.join(format!("{}-v1{}", layer_file_name, generation.get_suffix()))
```

Comment:
(This is more of a side note / slight nudge :) )

It's not obvious that there is a trailing `-` after the `-v1`.

Wondering if we're ever going to clean up the `Generation::None` variant

https://github.com/neondatabase/neon/blob/b655c7030ff2172e32f0c6e2e056183aa3b70d81/libs/utils/src/generation.rs#L12-L20

Reply:
We should start making it mandatory in external APIs in the near future (perhaps end of Q2/start of Q3).  We will still have to carry it in the index_part though, because old generation-less layers will exist indefinitely until some future point where we implement a format-modernizing rewriting scrub.